shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/EventBridgeOidcClient.java

```diff
@@ -0,0 +1,80 @@
+package com.redhat.service.bridge.shard.operator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.redhat.service.bridge.infra.auth.AbstractOidcClient;
+import com.redhat.service.bridge.infra.auth.OidcClientConfigUtils;
+import io.quarkus.oidc.client.OidcClientConfig;
+import io.quarkus.oidc.client.OidcClients;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.scheduler.Scheduled;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class EventBridgeOidcClient extends AbstractOidcClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventBridgeOidcClient.class);
+    private static final String NAME = "event-bridge-sso";
+
+    @ConfigProperty(name = "event-bridge.auth-server-url")
+    String serverUrl;
+
+    @ConfigProperty(name = "event-bridge.client-id")
+    String clientId;
+
+    @ConfigProperty(name = "event-bridge.credentials.secret")
+    String secret;
+
+    @ConfigProperty(name = "event-bridge.grant.type")
+    String type;
+
+    @ConfigProperty(name = "event-bridge.grant-options.password.username")
+    String username;
+
+    @ConfigProperty(name = "event-bridge.grant-options.password.password")
+    String password;
+
+    @Inject
+    public EventBridgeOidcClient(OidcClients oidcClients) {
+        super(NAME, oidcClients);
+    }
+
+    @PostConstruct
+    void init() {
+        Map<String, Map<String, String>> grantOptions = new HashMap<>();
+        Map<String, String> passwordConfig = new HashMap<>();
+        passwordConfig.put(OidcConstants.PASSWORD_GRANT_USERNAME, username);
+        passwordConfig.put(OidcConstants.PASSWORD_GRANT_PASSWORD, password);
+        grantOptions.put(OidcConstants.PASSWORD_GRANT, passwordConfig);
+
+        OidcClientConfig oidcClientConfig = new OidcClientConfig();
+        oidcClientConfig.setId(NAME);
+        oidcClientConfig.setAuthServerUrl(serverUrl);
+        oidcClientConfig.setClientId(clientId);
+        oidcClientConfig.grant.setType(OidcClientConfigUtils.getGrantType(type));
+        oidcClientConfig.setGrantOptions(grantOptions);
+        oidcClientConfig.getCredentials().setSecret(secret);
+        oidcClientConfig.setRefreshTokenTimeSkew(AbstractOidcClient.REFRESH_TOKEN_TIME_SKEW);
+
+        try {
+            this.init(oidcClientConfig);
+        } catch (RuntimeException e) {
+            LOGGER.error(String.format("Could not initialize OIDC client '%s'. The application is going to be stopped.", NAME), e);
+            Quarkus.asyncExit(1);
+        }
+    }
+
+    @Scheduled(every = AbstractOidcClient.SCHEDULER_TIME)
```

Comment:
@r00ta Can this be on the super-class or does the super-class need to be annotated as a managed bean for CDI?

Reply:
I've tried at the very beginning but I don't think it's possible due to 
```
java.lang.IllegalAccessError: class com.redhat.service.bridge.shard.operator.EventBridgeOidcClient_ScheduledInvoker_scheduledLoop_53cbd8a7d9f3118a28e39f47f0e83940b8366bbb tried to access method 'void com.redhat.service.bridge.infra.auth.AbstractOidcClient.scheduledLoop()' (com.redhat.service.bridge.shard.operator.EventBridgeOidcClient_ScheduledInvoker_scheduledLoop_53cbd8a7d9f3118a28e39f47f0e83940b8366bbb and com.redhat.service.bridge.infra.auth.AbstractOidcClient are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @47ad69f7)
	at com.redhat.service.bridge.shard.operator.EventBridgeOidcClient_ScheduledInvoker_scheduledLoop_53cbd8a7d9f3118a28e39f47f0e83940b8366bbb.invokeBean(EventBridgeOidcClient_ScheduledInvoker_scheduledLoop_53cbd8a7d9f3118a28e39f47f0e83940b8366bbb.zig:46)
	at io.quarkus.arc.runtime.BeanInvoker.invoke(BeanInvoker.java:20)
	at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$1.run(SimpleScheduler.java:274)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:548)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
I'd say if `Scheduled` is defined in another module for security reasons it fails with `com.redhat.service.bridge.infra.auth.AbstractOidcClient are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader `